File: core/src/ru/myitschool/dcrawler/encounters/DartEncounter.java
package ru.myitschool.dcrawler.encounters;

import ru.myitschool.dcrawler.effects.BloodiedEffect;
import ru.myitschool.dcrawler.entities.Entity;
import ru.myitschool.dcrawler.math.NumberAction;

/**
 * Created by Voyager on 18.02.2018.
 */
public class DartEncounter extends Encounter {

    public DartEncounter() {
        setName("Dart trap");
        setText("You was shoot by trap and now you bleed");
    }

    @Override
    protected void activate(Entity entity) {
        entity.addEffect(new BloodiedEffect(entity, new NumberAction(1), 3));
    }
}

File: core/src/com/wolg_vlad/dcrawler/encounters/DartEncounter.kt
package com.wolg_vlad.dcrawler.encounters

import com.wolg_vlad.dcrawler.effects.BloodiedEffect
import com.wolg_vlad.dcrawler.entities.Entity
import com.wolg_vlad.dcrawler.math.NumberAction

/**
 * Created by Voyager on 18.02.2018.
 */
class DartEncounter : Encounter() {
    override fun activate() {
        entity!!.addEffect(BloodiedEffect(entity, NumberAction(1), 3))
    }

    init {
        name = "Dart trap"
        text = "You was shoot by trap and now you bleed"
    }
}